### PR TITLE
Update the node docs

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

This fixes a few issues:

- https://github.com/Project-OSRM/osrm-backend/issues/3981
- There were updates to the `API.md` that were not reflected in the `node_osrm.cpp` source
- Adds better docs for all parameters of the constructor

## Tasklist
 - [x] review
 - [x] adjust for comments

## Requirements / Relations
 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/nodejs/node_osrm.cpp [hunk 1]
```diff
@@ -50,43 +50,39 @@ NAN_MODULE_INIT(Engine::Init)
...
- * | [`osrm.table`](#table)      | computes distance tables for given coordinates            |
- * | [`osrm.match`](#match)      | matches given coordinates to the road network             |
- * | [`osrm.trip`](#trip)        | computes the shortest trip between given coordinates      |
- * | [`osrm.tile`](#tile)        | Return vector tiles containing debugging info             |
+ * @param {Object} [options={shared_memory: true}] Options for creating an OSRM object.
+ * @param {string} [options.algorithm] The algorithm to use for routing. Can be 'CH', 'CoreCH' or 'MLD'.
+ *        Make sure you prepared the dataset with the correct toolchain.
+ * @param {bool} [options.shared_memory] Connects to the persistent shared memory datastore.
+ *        This requires you to run `osrm-datastore` prior to creating an `OSRM` object.
+ * @param {string} [options.path] The path to the `.osrm` files. This is mutually exclusive with setting {options.shared_memory} to true.
```
Comment (daniel-j-h) [inline src/nodejs/node_osrm.cpp]: `{options.shared_memory}` is not getting rendered properly

File: src/nodejs/node_osrm.cpp [hunk 2]
```diff
@@ -50,43 +50,39 @@ NAN_MODULE_INIT(Engine::Init)
...
- * | [`osrm.route`](#route)      | shortest path between given coordinates                   |
- * | [`osrm.nearest`](#nearest)  | returns the nearest street segment for a given coordinate |
- * | [`osrm.table`](#table)      | computes distance tables for given coordinates            |
- * | [`osrm.match`](#match)      | matches given coordinates to the road network             |
- * | [`osrm.trip`](#trip)        | computes the shortest trip between given coordinates      |
- * | [`osrm.tile`](#tile)        | Return vector tiles containing debugging info             |
+ * @param {Object} [options={shared_memory: true}] Options for creating an OSRM object.
+ * @param {string} [options.algorithm] The algorithm to use for routing. Can be 'CH', 'CoreCH' or 'MLD'.
+ *        Make sure you prepared the dataset with the correct toolchain.
+ * @param {bool} [options.shared_memory] Connects to the persistent shared memory datastore.
```
Comment (daniel-j-h) [inline src/nodejs/node_osrm.cpp]: `{bool}` should be `{Boolean}`

File: src/nodejs/node_osrm.cpp [hunk 3]
```diff
@@ -50,43 +50,39 @@ NAN_MODULE_INIT(Engine::Init)
...
- * | Service                     | Description                                               |
- * |-----------------------------|-----------------------------------------------------------|
- * | [`osrm.route`](#route)      | shortest path between given coordinates                   |
- * | [`osrm.nearest`](#nearest)  | returns the nearest street segment for a given coordinate |
- * | [`osrm.table`](#table)      | computes distance tables for given coordinates            |
- * | [`osrm.match`](#match)      | matches given coordinates to the road network             |
- * | [`osrm.trip`](#trip)        | computes the shortest trip between given coordinates      |
- * | [`osrm.tile`](#tile)        | Return vector tiles containing debugging info             |
+ * @param {Object} [options={shared_memory: true}] Options for creating an OSRM object.
+ * @param {string} [options.algorithm] The algorithm to use for routing. Can be 'CH', 'CoreCH' or 'MLD'.
```
Comment (daniel-j-h) [inline src/nodejs/node_osrm.cpp]: What's the default here?

File: docs/nodejs/api.md [hunk 4]
```diff
@@ -265,7 +303,256 @@ osrm.trip(options, function(err, response) {
...
+    -   `options.geometries` **[String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)?** Returned route geometry format (influences overview and per step). Can also be `geojson`. (optional, default `polyline`)
+    -   `options.overview` **[String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)?** Add overview geometry either `full`, `simplified` according to highest zoom level it could be display on, or not at all (`false`). (optional, default `simplified`)
+    -   `options.continue_straight` **[Boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)?** Forces the route to keep going straight at waypoints and don't do a uturn even if it would be faster. Default value depends on the profile.
+                         `null`/`true`/`false`
+-   `or` **[Boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** {Array} [options.annotations=false] Return annotations for each route leg.
```
Comment (daniel-j-h) [inline docs/nodejs/api.md]: This gets messed up in the rendered docs

**Comment by TheMarex** [comment]
@daniel-j-h Please have a quick look again. I think I caught all of them.

File: src/nodejs/node_osrm.cpp [hunk 5]
```diff
@@ -50,43 +50,39 @@ NAN_MODULE_INIT(Engine::Init)
...
- * The documentation on `osrm-extract` and `osrm-contract` for more information.
- * Once you have a complete `network.osrm` file, you can calculate routes in javascript with this library using the methods below.
- * To create an OSRM instance with your network you need to construct an instance like this:
+ * You can create such a `.osrm` file by running the OSRM binaries we ship in `node_modules/osrm/lib/binding/` and default
+ * profiles (e.g. for setting speeds and determining road types to route on) in `node_modules/osrm/profiles/`:
+ *
+ *     node_modules/osrm/lib/binding/osrm-extract data.osm.pbf -p node_modules/osrm/profiles/car.lua
+ *     node_modules/osrm/lib/binding/osrm-contract data.osrm
+ *
+ * Consult the [osrm-backend](https://github.com/Project-OSRM/osrm-backend) documentation or further details.
```
Comment (daniel-j-h) [inline src/nodejs/node_osrm.cpp]: or => for

**Pull request merged**